src/Coordination/KeeperStorage.cpp

```diff
@@ -1146,23 +1655,33 @@ KeeperStorage::ResponsesForSessions KeeperStorage::processRequest(const Coordina
         KeeperStorageRequestProcessorPtr request_processor = KeeperStorageRequestProcessorsFactory::instance().get(zk_request);
         Coordination::ZooKeeperResponsePtr response;
 
-        if (check_acl && !request_processor->checkAuth(*this, session_id))
+        if (is_local)
         {
-            response = zk_request->makeResponse();
-            /// Original ZooKeeper always throws no auth, even when user provided some credentials
-            response->error = Coordination::Error::ZNOAUTH;
+            assert(!zk_request->isReadRequest());
```

Comment:
```suggestion
            assert(zk_request->isReadRequest());
```

Reply:
??? :)

Reply:
now I'm confused, shouldn't the suggested change be correct? we need to assert that it's read request if it's locally processed. 😄